lib/Option/SanitizerOptions.cpp

```diff
@@ -34,6 +34,68 @@ static StringRef toStringRef(const SanitizerKind kind) {
   llvm_unreachable("Unsupported sanitizer");
 }
 
+llvm::SanitizerCoverageOptions swift::parseSanitizerCoverageArgValue(
+    const llvm::opt::Arg *A, const llvm::Triple &Triple,
+    DiagnosticEngine &Diags, SanitizerKind sanitizer) {
+
+  llvm::SanitizerCoverageOptions opts;
+  // The coverage names here follow the names used by clang's
+  // ``-fsanitize-coverage=`` flag.
+  for (int i = 0, n = A->getNumValues(); i != n; ++i) {
+    if (opts.CoverageType == llvm::SanitizerCoverageOptions::SCK_None) {
+      opts.CoverageType =
+          llvm::StringSwitch<llvm::SanitizerCoverageOptions::Type>(
+              A->getValue(i))
+              .Case("func", llvm::SanitizerCoverageOptions::SCK_Function)
+              .Case("bb", llvm::SanitizerCoverageOptions::SCK_BB)
+              .Case("edge", llvm::SanitizerCoverageOptions::SCK_Edge)
+              .Default(llvm::SanitizerCoverageOptions::SCK_None);
+      if (opts.CoverageType != llvm::SanitizerCoverageOptions::SCK_None)
+        continue;
+    }
+
+    if (StringRef(A->getValue(i)) == "indirect-calls") {
+      opts.IndirectCalls = true;
+      continue;
+    } else if (StringRef(A->getValue(i)) == "trace-bb") {
+      opts.TraceBB = true;
+      continue;
+    } else if (StringRef(A->getValue(i)) == "trace-cmp") {
+      opts.TraceCmp = true;
+      continue;
+    } else if (StringRef(A->getValue(i)) == "8bit-counters") {
+      opts.Use8bitCounters = true;
+      continue;
+    }
+
+    // Argument is not supported.
+    Diags.diagnose(SourceLoc(), diag::error_unsupported_option_argument,
+                   A->getOption().getPrefixedName(), A->getValue(i));
+    return llvm::SanitizerCoverageOptions();
+  }
+
+  if (opts.CoverageType == llvm::SanitizerCoverageOptions::SCK_None) {
+    Diags.diagnose(SourceLoc(), diag::error_option_missing_required_argument,
+                   A->getSpelling(),
+                   "The required argument is \"func\", \"bb\" or \"edge\"");
```

Comment:
Still not happy here; see the comment on the unsquashed commit https://github.com/delcypher/swift/commit/8f18b89e845a21ecdc54a4fe154a9a2c588fd571.